File: Backend/dgulaf/src/main/java/com/dgulaf/model/ItemLocation.java
package com.dgulaf.model;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "ItemLocation")
public class ItemLocation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer locationId;

    @ManyToOne
    @JoinColumn(name = "classroom_id", nullable = false)
    private Classroom classroom;

    @Column(length = 255)
    private String detail;
}

File: Frontend/dgu_laf/lib/model/item_location.dart
class ItemLocation {
  final int locationId;
  final int classroomId;
  final String? detail;

  ItemLocation({
    required this.locationId,
    required this.classroomId,
    this.detail,
  });

  factory ItemLocation.fromJson(Map<String, dynamic> json) {
    return ItemLocation(
      locationId: json['location_id'],
      classroomId: json['classroom_id'],
      detail: json['detail'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'location_id': locationId,
      'classroom_id': classroomId,
      'detail': detail,
    };
  }
}
